### DISCUSSION TITLE
Auto Generate Short Codes

### DISCUSSION THREAD

**Discussion by tony1661** [nonsalant/contract#26] [Ideas]
Hi,

I've implemented a feature in my lab system to automatically generate the short codes based on the HTML in `.ql-editor`.

This way I can paste in a template we use and generate the short codes automatically without me needing to remember which ones are in the HTML.

Is this a feature you'd be open to consider as a feature?

**Comment by nonsalant** [comment]
Hey Tony,

Sounds interesting — I think I’ve considered doing something similar but had some minor hang ups about it (besides the lack of time which wasn’t as minor). 

Would this make the square brackets around the shortcodes mandatory? 

This would mean square brackets can’t be used in the text anymore. Not sure how big of a deal this would be (no one being able to use the literal “[“ and “]” characters in their contracts anymore) and if would be worth doing double square brackets for the shortcodes  [[🤔]] 

I agree a version of this could be useful, though.

**Reply by tony1661** [reply]
@nonsalant the way I have it working currently is to scan the contract as is and populate the short codes. Theoretically if someone had [something] in their contract, they could remove the short code before generating.

**Reply by nonsalant** [reply]
Would you be able to make it use [[double square brackets]] as the new convention for the shortcodes? 

I wouldn’t want anyone to need to fiddle with the shortcodes panel below if they’re not using them (or they’re not aware they’re using them). E.g. if someone wants to add a footnote in their text and mark it [1] they may not be aware a shortcode field got filled out for them. 

Separately, the generator should also leave the shortcode alone if there’s no value for it, but I can change this part later.

**Reply by nonsalant** [reply]
> Would you be able to make it use [[double square brackets]] as the new convention for the shortcodes? 

@tony1661 -- so would you like to submit a pull request for this?
